code/code/disc/disc_wrath.cc

```diff
@@ -120,21 +119,28 @@ int plagueOfLocusts(TBeing *caster, TBeing *victim, int level, short bKnown)
     aff.bitvector = 0;
     locusts->affectTo(&aff);
 
+    locusts->setFighting(victim, 0, FALSE);
+
+    if (!victim->fight()) 
+      victim->setFighting(locusts, 0, FALSE);
+      
     return SPELL_SUCCESS;
   } else {
-    locusts->setLevel(WARRIOR_LEVEL_IND, (byte)lvl);
-    locusts->setHPLevel(lvl);
+    locusts->setLevel(WARRIOR_LEVEL_IND, (byte)normalizedLevel);
+    locusts->setHPLevel(normalizedLevel);
     locusts->setHPFromHPLevel();
-    locusts->setACLevel(lvl);
+    locusts->setACLevel(normalizedLevel);
     locusts->setACFromACLevel();
-    locusts->setDamLevel(lvl);
+    locusts->setDamLevel(normalizedLevel);
```

Comment:
Why the duplicate code? Looks like it could be extracted from these branches.

Reply:
I had the same thought then just forgot about it. I'll take another look at it.

Reply:
I remember now why I left it - it was because of the crit success/isLucky saving throw code that @wershlak mentioned above as being unnecessary. On prayer success there was a chance that the locusts' level would be doubled or halved, so we needed to apply those modifiers in each branch.

Reply:
Fixed in https://github.com/sneezymud/sneezymud/pull/353/commits/bb6987a12f3deb50ecb985e9095d18fe80652d58, though I'm not sure why the check is failing for that commit